src/models/scatter.js

```diff
@@ -448,20 +451,22 @@ nv.models.scatter = function() {
                     .type(function(d) { return getShape(d[0]); })
                     .size(function(d) { return z(getSize(d[0],d[1])) })
             );
-            points.exit().remove();
+            points.exit().each(delCache).remove();
             groups.exit().selectAll('path.nv-point')
                 .watchTransition(renderWatch, 'scatter exit')
                 .attr('transform', function(d) {
                     return 'translate(' + nv.utils.NaNtoZero(x(getX(d[0],d[1]))) + ',' + nv.utils.NaNtoZero(y(getY(d[0],d[1]))) + ')'
                 })
                 .remove();
-            points.filter(function (d) { return scaleDiff || sizeDiff || getDiffs(d, 'x', 'y'); })
+            // Update points position only if "x" or "y" have changed
+            points.filter(function (d) { return scaleDiff || sizeDiff || getDiffs(d, 'x', getX, 'y', getY); })
```

Comment:
> You could use function.toString() as the key.
Yes this is the ES5 option I meant but it's it's more performance sensitive not only calling yet another function to get a potentially huge string but also using huge strings as keys in a hash table.

> Or if you really want them named, just pass it as an object, so like:
> 
> {x: getX, y: getY}, so that there is a clear relation instead of the next dev having to parse through the code to figure out how those arguments relate.
I agree that would be clearer but adding (point * attributes) of throwaway objects every update is a lot of work for the GC. objects are particularly expensive to create and destroy in large numbers, where as the scope and other primitives inside the function calls are cheap. (only objects are currently the cache entries which persist across updates).

I'm on board with making the code clearer but I think it needs to not undermine the purpose of it. I'm open to suggestions if you can think of any better ways, alternatively I can comment the gettDiffs function to make it's purpose and interface more explicitly apparent.

If I had to choose one of your current options I'd go for function.toString() because that probably has less overhead and would be smoother than GC which operates intermittently.